packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -386,6 +377,8 @@ function patchCustomElement(
     parent: ParentNode,
     renderer: RendererAPI
 ) {
+    // TODO [#3331]: this check and the corresponding code block is only needed for lwc:dynamic.
+    // It can be safely removed with the directive.
```

Comment:
This can be removed in 246 once `lwc:dynamic` is deprecated, it's only use was for `lwc:dynamic`.